presto-main/src/main/java/io/prestosql/security/ConnectorAccessControlForProcedures.java

```diff
@@ -0,0 +1,354 @@
+import io.prestosql.spi.connector.SchemaRoutineName;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.security.PrestoPrincipal;
+import io.prestosql.spi.security.Privilege;
+import io.prestosql.spi.security.ViewExpression;
+import io.prestosql.spi.type.Type;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Iterables.getFirst;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorAccessControlForProcedures
+        implements ConnectorAccessControl
+{
+    private final AccessControl accessControl;
+    private final SecurityContext securityContext;
+    private final String catalogName;
+
+    public ConnectorAccessControlForProcedures(AccessControl accessControl, SecurityContext securityContext, String catalogName)
+    {
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.securityContext = requireNonNull(securityContext, "securityContext is null");
+        this.catalogName = requireNonNull(catalogName, "catalogName is null");
+    }
+
+    @Override
+    public void checkCanCreateSchema(ConnectorSecurityContext context, String schemaName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanCreateSchema(securityContext, getCatalogSchemaName(schemaName));
+    }
+
+    @Override
+    public void checkCanDropSchema(ConnectorSecurityContext context, String schemaName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanDropSchema(securityContext, getCatalogSchemaName(schemaName));
+    }
+
+    @Override
+    public void checkCanRenameSchema(ConnectorSecurityContext context, String schemaName, String newSchemaName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanRenameSchema(securityContext, getCatalogSchemaName(schemaName), newSchemaName);
+    }
+
+    @Override
+    public void checkCanSetSchemaAuthorization(ConnectorSecurityContext context, String schemaName, PrestoPrincipal principal)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanSetSchemaAuthorization(securityContext, getCatalogSchemaName(schemaName), principal);
+    }
+
+    @Override
+    public void checkCanShowSchemas(ConnectorSecurityContext context)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanShowSchemas(securityContext, catalogName);
+    }
+
+    @Override
+    public Set<String> filterSchemas(ConnectorSecurityContext context, Set<String> schemaNames)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
```

Comment:
`        checkArgument(context == null, "context should be null");` Here and elsewhere